common/src/main/java/io/netty5/util/concurrent/DefaultTaskScheduler.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2023 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty5.util.concurrent;
+
+import io.netty5.util.concurrent.AbstractScheduledEventExecutor.RunnableScheduledFutureNode;
+import io.netty5.util.internal.DefaultPriorityQueue;
+import io.netty5.util.internal.PriorityQueue;
+
+import java.util.Comparator;
+import java.util.concurrent.Callable;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Default task scheduler for {@link AbstractScheduledEventExecutor} that want to support task scheduling.
+ * Uses {@link DefaultPriorityQueue} to schdeule tasks.
+ */
+public final class DefaultTaskScheduler extends AbstractTaskScheduler {
+
+    private static final Comparator<RunnableScheduledFutureNode<?>> SCHEDULED_FUTURE_TASK_COMPARATOR =
+            Comparable::compareTo;
```

Comment:
**Performance Issue**: The use of Comparable::compareTo in the DefaultTaskScheduler may not be the most efficient way to compare RunnableScheduledFutureNode instances. This approach relies on the natural ordering of the elements, which might not be optimized for the specific use case of task scheduling, potentially leading to performance degradation when managing large numbers of scheduled tasks. <br> **Fix**: Implement a custom Comparator for RunnableScheduledFutureNode instances that is specifically optimized for the task scheduling use case. This custom Comparator should prioritize the most critical factors for task ordering (e.g., execution time) to ensure efficient task scheduling and execution. <br> **Code Suggestion**: 
 ```
 private static final Comparator<RunnableScheduledFuture<?>> TASK_SCHEDULER_COMPARATOR = new Comparator<RunnableScheduledFuture<?>>() {
     @Override
     public int compare(RunnableScheduledFuture<?> o1, RunnableScheduledFuture<?> o2) {
         return Long.compare(o1.getDelay(TimeUnit.NANOSECONDS), o2.getDelay(TimeUnit.NANOSECONDS));
     }
 };

 private PriorityQueue<RunnableScheduledFuture<?>> scheduledTaskQueue = new PriorityQueue<>(TASK_SCHEDULER_COMPARATOR);
 ```